app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,36 @@
 class PostsController < ApplicationController
+  def new
+    @post = Posts.new(author_id: params[:user_id])
+  end
+
   def index
+    user_id = params[:user_id]
+    @posts = Users.where(id: user_id)[0].get_all_posts(user_id)
```

Comment:
- [ ] I suggest that you solve N+1 problem to be able to fetch all posts and their comments for a user you can simply do
`@posts = @user.posts.includes(:comments)`